airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/util/S3StreamTransferManagerHelper.java

```diff
@@ -26,19 +26,61 @@
 
 import alex.mojaki.s3upload.StreamTransferManager;
 import com.amazonaws.services.s3.AmazonS3;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class S3StreamTransferManagerHelper {
 
+  protected static final Logger LOGGER = LoggerFactory.getLogger(S3StreamTransferManagerHelper.class);
+
   // See this doc about how they affect memory usage:
   // https://alexmojaki.github.io/s3-stream-upload/javadoc/apidocs/alex/mojaki/s3upload/StreamTransferManager.html
   // Total memory = (numUploadThreads + queueCapacity) * partSize + numStreams * (partSize + 6MB)
   // = 31 MB at current configurations
   public static final int DEFAULT_UPLOAD_THREADS = 2;
   public static final int DEFAULT_QUEUE_CAPACITY = 2;
   public static final int DEFAULT_PART_SIZE_MB = 5;
+  // MAX object size for AWS and GCS is 5TB (max allowed 10,000 parts*525mb)
+  // (https://aws.amazon.com/s3/faqs/, https://cloud.google.com/storage/quotas)
+  public static final int MAX_ALLOWED_PART_SIZE_MB = 525;
   public static final int DEFAULT_NUM_STREAMS = 1;
 
-  public static StreamTransferManager getDefault(String bucketName, String objectKey, AmazonS3 s3Client) {
+  public static StreamTransferManager getDefault(String bucketName, String objectKey, AmazonS3 s3Client, Long partSize) {
+    if (partSize == null) {
+      LOGGER.warn(String.format("Part side for StreamTransferManager is not set explicitly. Will use the default one = %sMB. "
```

Comment:
it looks like a mistyped `side` instead of `size`

Reply:
updated, thanks